### PR TITLE
Don't start daemon in userns mode if graphdir inaccessible

### LINKED ISSUE #25847: Do we want to (and will users accept) the Docker daemon changing permissions on an existing path for (root dir) access to work via a user namespaced process?
Just as Ubuntu 16.04 [lxcfs bug](https://github.com/docker/docker/issues/24834#issuecomment-240836035) got fixed, now with the new namespace feature we are not able to use namespaces together with the `--graph` parameter.

When using Docker 1.12.0 and lxcfs 2.0.3 with `--userns-remap=Default` and `--graph /media/user/somedrive/docker` throws the following error when doing any command such as `docker run hello-world`:

`docker: Error response from daemon: oci runtime error: rootfs_linux.go:42: preparing rootfs caused "permission denied".`

On my development machine I have a small SSD so I'm not able to use the default `/var/lib/docker` for long before I run out of space from having to work with many images. Being able to use the `--graph` feature would really help with the already awesome docker namespace feature.

### PR DESCRIPTION
**Pull request by estesp**
Fixes #25847 

Warn the user and fail daemon start if the graphdir path has any
elements which will deny access to the remapped root uid/gid.

Docker-DCO-1.1-Signed-off-by: Phil Estes estesp@linux.vnet.ibm.com

### CODE REVIEW

**Comment by estesp** [comment]
ping @justincormack per your comment in the issue noted--this errors out after checking world "x" (or ownership/group access).

**Comment by justincormack** [comment]
LGTM

**Comment by vdemeester** [comment]
LGTM 🐮

File: pkg/idtools/idtools_unix.go [hunk 1]
```diff
@@ -58,3 +58,29 @@ func mkdirAs(path string, mode os.FileMode, ownerUID, ownerGID int, mkAll, chown
 	}
 	return nil
 }
+
+// CanAccess takes a valid (existing) directory and a uid, gid pair and determines
+// if that uid, gid pair has access (execute bit) to the directory
+func CanAccess(path string, uid, gid int) bool {
+	statInfo, err := system.Stat(path)
```
Comment (justincormack) [inline pkg/idtools/idtools_unix.go]: os.Stat not system.Stat

**Comment by justincormack** [comment]
Janky is complaining

**Comment by estesp** [comment]
Interesting--looks like I'm the first user of docker/pkg/system.Stat from common code that is built for darwin; will fix

**Comment by vdemeester** [comment]
LGTM :cat:

**Pull request merged**